.github/workflows/publish-img.yaml

```diff
@@ -40,13 +51,15 @@ jobs:
           username: ${{ github.actor }}
           password: ${{ secrets.GITHUB_TOKEN }}
 
-      - name: Push latest container image
-        if: github.repository_owner == 'kubevirt'
+      - name: Push branch-based and latest container images
+        if: github.repository_owner == 'kubevirt' && github.ref_type == 'branch'
         uses: docker/build-push-action@v5.3.0
         with:
           context: .
           push: true
-          tags: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}:latest
+          tags: |
+            ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}:latest
```

Comment:
wait, iiuc - I might be wrong - this means that pushing to a branch will also overwrite the latest img tag. We don't want that.

We only want to update the `latest` img tag on pushes to the "main" branch.

Reply:
No, I prefer to see everything at once, otherwise I don't understand what's happening. 

This file is for the main branch *or* for tags named `vX.Y.Z` . You're adding a condition which I think skips the tags, hence this will only apply for the main branch. 

Thus, I don't understand what you're fixing in this PR. The main branch is kind of OK, right ? We're simply missing configurations for the other branches.